stf.test/build.gradle

```diff
@@ -21,37 +25,59 @@ dependencies {
     // TODO: The custom saros eclipse plugin adds all dependencies to implementation
     stfTestCompile configurations.implementation
     testCompile configurations.stfTestCompile // for running the tests in eclipse
-}
 
-// TODO move into buildSrc
-task stfTest (type: Test, dependsOn: [':saros.eclipse:build']) {
-    group 'Verification'
-    description 'Runs the stf tests. Requires a corresponding test environment'
-
-    testClassesDirs = sourceSets.stfTest.output.classesDirs
+    stfFlakyAndFailingTestCompile project(':saros.stf')
+    stfFlakyAndFailingTestCompile project(path: ':saros.stf', configuration: 'testing')
+    stfFlakyAndFailingTestCompile configurations.testConfig
+    stfFlakyAndFailingTestCompile configurations.implementation
+    testCompile configurations.stfFlakyAndFailingTestCompile // for running the tests in eclipse
+}
 
+class StfTest extends Test {
+  StfTest() {
+    this.setGroup('Verification')
+    this.setDescription('Runs the stf tests. Requires a corresponding test environment')
     systemProperty 'saros.stf.client.configuration.files', System.getProperty('stf.client.configuration.files', '')
 
-    classpath = sourceSets.stfTest.runtimeClasspath
-
-    testLogging {
-        showStandardStreams = true
-        events "started", "passed", "skipped", "failed", "standardOut", "standardError"
+    this.testLogging{ loggingContainer ->
+        loggingContainer.setShowStandardStreams(true)
+        loggingContainer.setEvents(['started', 'passed', 'skipped', 'failed', 'standardOut', 'standardError'])
     }
-    // Print summary of failed tests
-    ext.failedStfTests = []
-    afterTest { descriptor, result ->
-        if(result.resultType == org.gradle.api.tasks.testing.TestResult.ResultType.FAILURE){
-            ext.failedStfTests << "${descriptor.className}.${descriptor.name}"
+
+    List<String> failedTests = new ArrayList<String>();
+    this.afterTest { descriptor, result ->
+        if( result.resultType == org.gradle.api.tasks.testing.TestResult.ResultType.FAILURE ){
+            failedTests << "${descriptor.className}.${descriptor.name}"
         }
     }
-    afterSuite { suite, result ->
-        if (!suite.parent && !ext.failedStfTests.isEmpty()) {
+
+    this.afterSuite { suite, result ->
+        if ( !suite.parent && !failedTests.isEmpty() ) {
             logger.lifecycle("Failed tests:")
-            ext.failedStfTests.each { logger.lifecycle(it) }
+            failedTests.each { logger.lifecycle(it) }
         }
     }
 
-    include '**/stf/test/**/*Test.*'
-    exclude '**/stf/test/stf/**'
+    this.include('**/stf/test/**/*Test.*')
+    this.exclude('**/stf/test/stf/**')
+  }
 }
+
+task stfTest (type: StfTest, dependsOn: [':saros.eclipse:build']) {
```

Comment:
Isn't there a better way, or to be more specific - is it possible to not use 2 different directories and just tell the JUNIT runner to skip all test cases that are NOT annotated with a specific category ?

Reply:
What do mean with:
> not use 2 different directories

The stf tests (either failing, flaky or stable) are all in the same directory. But I will try to reduce the redundancy of the `stfTest`/`stfFlakyAndFailingTest` definition (I think this looks like the tests are located in different directories)

> and just tell the JUNIT runner to skip all test cases that are NOT annotated with a specific category ?

This is what I do in:
```gradle
useJUnit {
        excludeCategories 'saros.stf.test.categories.FlakyTests', 'saros.stf.test.categories.FailingTests'
    }
```

Reply:
I just got confused wit the testClassesDirs on line 76. But it seems the definitions containing only the same directory.